flaky/_flaky_plugin.py

```diff
@@ -138,28 +177,21 @@ def _handle_test_error_or_failure(self, test, err):
         except AttributeError:
             return False
 
-        need_reruns, should_rerun = self._should_handle_test_error_or_failure(
-            test,
-            name,
-            err,
-        )
-
         if self._has_flaky_attributes(test):
             self._add_flaky_test_failure(test, err)
-            flaky = self._get_flaky_attributes(test)
-            runs = flaky[FlakyNames.CURRENT_RUNS] + 1
-            self._set_flaky_attribute(test, FlakyNames.CURRENT_RUNS, runs)
-            if need_reruns:
+            should_handle = self._should_handle_test_error_or_failure(test)
+            self._increment_flaky_attribute(test, FlakyNames.CURRENT_RUNS)
+            if should_handle:
                 flaky = self._get_flaky_attributes(test)
```

Comment:
In this function you call it `flaky`, but in a function above you renamed the same thing from `flaky` to `flaky_attributes`.